### ISSUE TITLE
Trying to open a deleted repo from Team Explorer crashes VS

### ISSUE THREAD

**Issue by grokys** [github/VisualStudio#566]
On master, if I double click on a repo in the connect section that has been deleted from GitHub and then open the GitHub pane, Visual Studio has a hard crash with an `Octokit.NotFoundException`.

**Comment by jswietek** [comment]
Hello,

I'm looking into this issue. I fond where the exception is thrown, and found a way to handle it.
But, looking at the functionality: does it make any sense to keep the deleted repository in the GitHub section of the Connect pane?

Let me explain on an example, of how I'm reproducing the issue.
1) I created a project (named ConsoleApplication112) and a git repo for it. 
2) I published it to GitHub. 
3) Then, from GitHub website, I deleted this repo.
4) Then I created a new project (without git repo) and open TeamExplorer:

![image](https://cloud.githubusercontent.com/assets/8940080/20103303/af9f8fec-a5c9-11e6-819a-3e98b8cef090.png)

If the ConsoleApplication112 was deleted from github and I restarted VS, why this repo is still there?

**Comment by tocsoft** [comment]
I feel this crash is going to be similar to the `AuthorizationException` crash (fixed in #653) probably just needs an 
`.Catch<System.Reactive.Unit, Octokit.NotFoundException>( ... )` adding to the observable to swallow the `Octokit.NotFoundException`. Just want to make sure that they are **not** logged out on 'NotFoundException' which is the only difference between the 2 exceptions. 

I realise this doesn't solve the showing in the GitHub section but even if it was hidden from there I suspect it would still crash trying to do the PR list lookup if you opened any repo that used to be on github(still has a remote set) but has been removed from the site. 

Basically I see this as 2 issues;
 1. crashing when opening a repo that no longer exists on github, 
 2. Showing removed repositories in GitHub  section.

**Comment by jswietek** [comment]
@tocsoft 
Thanks for the reply. The fix looks exactly as you say.

I will create a PR for that one when I have some free time, as well as submit an issue with removed repos visible in github section.

**Comment by shana** [comment]
The repo is still on the list because that list is populated from the known repositories list in VS, and as far as VS is concerned, that repository exists locally so it's fine, and as far as the GitHub extension is concerned, that repository has a valid remote configured, so it's fine too.

Since the list is populated from the local registry and it doesn't strictly require a login to populate (the repos exist locally and therefore are known), it's hard to impossible to detect whether the repository doesn't exist locally because it got deleted, or it doesn't exist remotely because your current credentials don't allow you to access it (same result code on both cases)

We can't safely assume that it no longer exists remotely, so we can't take it out of the list, and we can't just randomly remove the remote configuration for it, for the same reason. What we can do is not crash (that's certainly a thing we shouldn't be doing), but not much else 😕

**Comment by jswietek** [comment]
@shana 
Thanks for the explanation, this clears some things for me.
Still I feel like it would make sense to remove the repository from the list after we tried (and failed) to login to it (or perform any other action on it e.g. like in this case - we get a NotFoundException when we try to get the pull requests).

Anyway, I'll take a look into possibility of that change :)

**Comment by shana** [comment]
@jswietek Yeah, I understand it can be a confusing edge case. You can only be logged in to github.com with one profile at a time, but you can have more than one account in certain conditions. This means that you may have forgotten to log out of another account when trying to open a private repository, so it will return a NotFound even though the repo is there, you're just on the wrong account. And the reality is, the repository exists locally, so you should be able to switch to it and open the solution just fine.

Maybe we should take it out of the list, maybe not, it's a thing we can discuss. Maybe not in this issue, as that's a separate design discussion from the actual crash that this issue is about, feel free to open a new issue to discuss that and link here for context 😄

**Comment by grokys** [comment]
This should now be fixed.
